### ISSUE TITLE
Can not reproduce the results on GLUE

### ISSUE THREAD

**Issue by Car-pe** [GCYZSL/MoLA#12]
Hello，
I use the dataset from this link "https://drive.google.com/file/d/1-AHDmTKnds9JTJPFr1CFCqneGjj1HB0M/view?usp=sharing" to reproduce your results on GLUE, but I got "acc: 0.0" on CoLA/MRPC/RTE.  

Following is my script to train and eval:
Train:
  CUDA_VISIBLE_DEVICES=0,1,2,3 python mola_training.py \
           --base_model "NousResearch/Llama-2-7b-hf" \
          --data_path "./datasets/datasets/glue_rte_all.hf" \
          --output_dir "./results/rte/0.0" \
           --batch_size 128 \
           --micro_batch_size 8 \
           --num_epochs 20 \
           --learning_rate 3e-4 \
           --cutoff_len 256 \
           --val_set_size 1 \
           --lora_r "8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8" \
           --lora_alpha 16 \
           --lora_dropout 0.05 \
           --lora_target_modules "q_proj,v_proj,k_proj,o_proj,gate_proj,down_proj,up_proj" \
           --number_experts "2,2,2,2,2,2,2,2,4,4,4,4,4,4,4,4,6,6,6,6,6,6,6,6,8,8,8,8,8,8,8,8" \
           --top_k "2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2" \
           --train_on_inputs \
           --group_by_length \
           --add_eos_token 
Eval:
  CUDA_VISIBLE_DEVICES=4,5,6,7 python evaluation_scienceqa.py \
          --base_model "NousResearch/Llama-2-7b-hf" \
          --test_dataset "./datasets/datasets/glue_rte_test.json" \
          --mola_weights "./results/rte/0.0" \
          --save_path "./results/rte/0.0/mola_test.json" \
          --batch_size 8 \
          --lora_target_modules "q_proj,v_proj,k_proj,o_proj,gate_proj,down_proj,up_proj" \
          --number_experts "2,2,2,2,2,2,2,2,4,4,4,4,4,4,4,4,6,6,6,6,6,6,6,6,8,8,8,8,8,8,8,8" \
          --top_k "2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2" \

**Comment by Car-pe** [comment]
And on scienceqa dataset, my result is "91.59", less than "92.36" claimed on the paper.

**Comment by GCYZSL** [comment]
Hi, evaluation_scienceqa.py is used to evaluate QA datasets. To evaluate for GLUE datasets, you need to modify the evaluation file by replacing the choices with the ground-truth labels of the corresponding tasks. As we mentioned in our paper, we did a grid search on the epochs to get the final result. Thank you!

**Comment by Car-pe** [comment]
ok，thank you，I will try it.
And do you mean evaluation_scienceqa.py can be used to evaluate the CommonsenseQA and OpenbookQA ?
Thank you!

**Comment by GCYZSL** [comment]
Due to the time limits, I have not tested this on other QA datasets, but I believe it can be used to evaluate QA tasks. Thank you.

**Comment by Car-pe** [comment]
These two QA datasets are claimed on your paper
<img width="717" alt="image" src="https://github.com/GCYZSL/MoLA/assets/87354341/8d780d1b-da1b-42b0-90ba-58fca1c9a08c">

**Comment by GCYZSL** [comment]
That’s what I mean.

**Comment by Car-pe** [comment]
Ok, thank you

**Comment by GCYZSL** [comment]
Thank you!

**Comment by Car-pe** [comment]
Hello，could you please send me the python files used to evaluate the GLUE datasets? Thank you!
My email is jianlong002@gmail.com

**Comment by GCYZSL** [comment]
Hello, could you try to change the line 200 in evaluation_scienceqa.py to:
`pattern = re.compile(r'Answer: ([^\.]*)')`
Thank you!
